src/python/grpcio_tests/tests_aio/unit/compression_test.py

```diff
@@ -0,0 +1,174 @@
+    except RuntimeError:
+        pass
+    else:
+        raise ValueError(
+            'Expecting exceptions if set_compression is not effective')
+
+
+async def _test_disable_compression_unary(request, context):
+    assert _REQUEST == request
+    context.set_compression(grpc.Compression.Deflate)
+    context.disable_next_message_compression()
+    return _RESPONSE
+
+
+async def _test_disable_compression_stream(unused_request_iterator, context):
+    assert _REQUEST == await context.read()
+    context.set_compression(grpc.Compression.Deflate)
+    await context.write(_RESPONSE)
+    context.disable_next_message_compression()
+    await context.write(_RESPONSE)
+    await context.write(_RESPONSE)
+
+
+_ROUTING_TABLE = {
+    _TEST_UNARY_UNARY:
+        grpc.unary_unary_rpc_method_handler(_test_unary_unary),
+    _TEST_SET_COMPRESSION:
+        grpc.stream_stream_rpc_method_handler(_test_set_compression),
+    _TEST_DISABLE_COMPRESSION_UNARY:
+        grpc.unary_unary_rpc_method_handler(_test_disable_compression_unary),
+    _TEST_DISABLE_COMPRESSION_STREAM:
+        grpc.stream_stream_rpc_method_handler(_test_disable_compression_stream),
+}
+
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def service(self, handler_call_details):
+        return _ROUTING_TABLE.get(handler_call_details.method)
+
+
+async def _start_test_server(options=None):
+    server = aio.server(options=options)
+    port = server.add_insecure_port('[::]:0')
+    server.add_generic_rpc_handlers((_GenericHandler(),))
+    await server.start()
+    return f'localhost:{port}', server
+
+
+class TestCompression(AioTestBase):
+
+    async def setUp(self):
+        server_options = (_GZIP_DISABLED_CHANNEL_ARGUMENT,)
+        self._address, self._server = await _start_test_server(server_options)
+        self._channel = aio.insecure_channel(self._address)
+
+    async def tearDown(self):
+        await self._channel.close()
+        await self._server.stop(None)
+
+    async def test_channel_level_compression(self):
+        # GZIP is disabled, this call should fail
+        async with aio.insecure_channel(
+                self._address, compression=grpc.Compression.Gzip) as channel:
+            multicallable = channel.unary_unary(_TEST_UNARY_UNARY)
+            call = multicallable(_REQUEST)
+            with self.assertRaises(aio.AioRpcError) as exception_context:
+                await call
+            rpc_error = exception_context.exception
+            self.assertEqual(grpc.StatusCode.UNIMPLEMENTED, rpc_error.code())
+
+        # Deflate is allowed, this call should succeed
+        async with aio.insecure_channel(
+                self._address, compression=grpc.Compression.Deflate) as channel:
+            multicallable = channel.unary_unary(_TEST_UNARY_UNARY)
+            call = multicallable(_REQUEST)
+            self.assertEqual(grpc.StatusCode.OK, await call.code())
+
+    async def test_client_call_level_compression(self):
```

Comment:
Same as before, I would split this test into two different tests with an explicit name.

Reply:
Split into two test cases.